### PR TITLE
Rewrote nested functions tour

### PR DESCRIPTION
**Pull request by ZeonTrevor**
an easier to understand example of using nested methods through the computation of factorial.

### CODE REVIEW

File: tutorials/tour/_posts/2017-02-13-nested-functions.md [hunk 1]
```diff
@@ -11,25 +11,27 @@ next-page: currying
...
 
 ```tut
-object FilterTest extends App {
-  def filter(xs: List[Int], threshold: Int) = {
-    def process(ys: List[Int]): List[Int] =
-      if (ys.isEmpty) ys
-      else if (ys.head < threshold) ys.head :: process(ys.tail)
-      else process(ys.tail)
-    process(xs)
+object FactorialTest extends App {
```
Comment (MasseGuillaume) [inline tutorials/tour/_posts/2017-02-13-nested-functions.md]: I would use `def main` here

**Comment by travis032654** [comment]
Hey @ZeonTrevor, huge thanks for your work. It's going to help a lot of people who use this as their primary reference. Check out the conversation here: #733 

I'm also tracking the work in the Project Page. You'll find the link on the right.

File: tutorials/tour/_posts/2017-02-13-nested-functions.md (hunk 1 above)
Comment (travis032654) [inline tutorials/tour/_posts/2017-02-13-nested-functions.md]: I think it's clearer if all of the examples are in worksheet mode and not inside classes.

Comment (ZeonTrevor) [inline tutorials/tour/_posts/2017-02-13-nested-functions.md]: @travis032654 I have rewritten the example in worksheet mode

**Comment by SethTisue** [comment]
nice, thanks!

**Comment by SethTisue** [comment]
after merging I noticed an error that I fixed in 4becbaf7f2ecbe98a1dbb2f013dac1f865c28142

**Pull request merged**